packages/firestore/android/src/reactnative/java/io/invertase/firebase/firestore/ReactNativeFirebaseFirestoreCollectionModule.java

```diff
@@ -193,6 +197,99 @@ public void collectionCount(
             });
   }
 
+  @ReactMethod
+  public void aggregateQuery(
+      String appName,
+      String databaseId,
+      String path,
+      String type,
+      ReadableArray filters,
+      ReadableArray orders,
+      ReadableMap options,
+      ReadableArray aggregateQueries,
+      Promise promise) {
+    FirebaseFirestore firebaseFirestore = getFirestoreForApp(appName, databaseId);
+    ReactNativeFirebaseFirestoreQuery firestoreQuery =
+        new ReactNativeFirebaseFirestoreQuery(
+            appName,
+            databaseId,
+            getQueryForFirestore(firebaseFirestore, path, type),
+            filters,
+            orders,
+            options);
+
+    ArrayList<AggregateField> aggregateFields = new ArrayList<>();
+
+    for (int i = 0; i < aggregateQueries.size(); i++) {
+      ReadableMap aggregateQuery = aggregateQueries.getMap(i);
+      String aggregateType = aggregateQuery.getString("aggregateType");
+      String fieldPath = aggregateQuery.getString("field");
+
+      switch (Objects.requireNonNull(aggregateType)) {
```

Comment:
similar to the iOS error handling comment - this will throw NullPointerException if for some reason it evaluates to null, and that is a separate error-handling pathway for calling code vs Promise.reject so it will be unexpected.

It seems like this is just to handle "may be null?" Android Studio warnings?
Let me have a look at reshaping this so the warnings go away, but the non-null type assertions / conversions for safety are done in a try/catch where the final error handling pathway is a Promise reject if any of them are null

I'll push a commit that has that change in a moment